src/ln/channelmanager.rs

```diff
@@ -1526,7 +1529,8 @@ impl ChannelManager {
 						rejected_by_dest: !payment_retryable,
 					});
 				} else {
-					panic!("should have onion error packet here");
+					//TODO (ariard) which failure code to generate from unilateral/revoked channel closing event ? currently NODE|2 (temporary_node_failure)
+					//panic!("should have onion error packet here");
```

Comment:
Dont need the TODO, this is wrong(ish), see-also https://github.com/rust-bitcoin/rust-lightning/pull/243/files#diff-d152f468921ba1eb6da8dac49a90a97aL1516

Reply:
Not yet corrected but seems doable to send back a ErrorPacket here as it generated after call to fetch_pending_htlc_updated in case of HTLC-timeout (need to agree on which error code to give back)

Reply:
ie you can replace the TODO with something like "TODO: Pass this back (see GH #243)"

Reply:
Ah yeah, so commented out because it's crashing my tests right now!